jax/_src/tree_util.py

```diff
@@ -46,56 +50,59 @@ def tree_flatten(tree, is_leaf: Optional[Callable[[Any], bool]] = None):
       stopped immediately, with the whole subtree being treated as a leaf.
 
   Returns:
-    A pair where the first element is a list of leaf values and the second
+    A pair where the first element is a sequence of leaf values and the second
```

Comment:
How about we keep it as returning a list? In general I think it makes sense for the input specification to be loose and the output specification to be tight. (We would update the above annotation too.)